tsec-libsodium/src/test/scala/tsec/libsodium/KeyDerivationTest.scala

```diff
@@ -0,0 +1,60 @@
+package tsec.libsodium
+
+import cats.data.StateT
+import tsec.common._
+import cats.effect.IO
+import tsec.libsodium.kdf.KeyDerivation
+
+class KeyDerivationTest extends SodiumSpec {
+
+  behavior of "Key derivation"
+
+  it should "generate multiple keys" in {
+
+    val program = for {
+      master <- KeyDerivation.generateKey[IO]
+      context = "Examples"
+      _ <- KeyDerivation.deriveKey[IO](master, 16, 1, context)
+      _ <- KeyDerivation.deriveKey[IO](master, 32, 2, context)
+      _ <- KeyDerivation.deriveKey[IO](master, 64, 3, context)
+    } yield ()
```

Comment:
What is this really testing? It doesn't assert anything about the derived keys other than that nothing blows up unexpectedly.

I'd expect (if I understand the API correctly) that

* `deriveKey` gives the same result for multiple invocations with the same arguments
* `deriveKey` gives different results (almost certainly) for a different context

Reply:
I have to agree here. These are the things we should be testing for re: key derivation.